extension/content/utils/api.js

```diff
@@ -1,30 +1,111 @@
-export async function* fetchRecipes(environment, qs = {}) {
-  let url = new URL(`${environment.getUrl()}api/v3/recipe/`);
-  for (const [key, value] of Object.entries(qs)) {
-    url.searchParams.set(key, value);
+export default class NormandyAPI {
+  constructor(environment) {
+    this.environment = environment;
   }
 
-  while (url) {
-    let res = await fetch(url);
-    let data = await res.json();
-    url = data.next;
-    for (const recipe of data.results) {
-      yield recipe;
+  getBaseUrl({ version = 3, mode }) {
+    const base =
+      mode.toLowerCase() === "w"
+        ? this.environment.writeableUrl
+        : this.environment.readOnlyUrl;
+
+    return new URL(`api/v${version}/`, base).href;
+  }
+
+  async request({ version = 3, url, extraHeaders, ...options }) {
+    const headers = new Headers();
+    headers.append("Accept", "application/json");
+    if (!(options.body && options.body instanceof FormData)) {
+      headers.append("Content-Type", "application/json");
+    }
+
+    for (let headerName in extraHeaders) {
+      headers.append(headerName, extraHeaders[headerName]);
+    }
+
+    const settings = {
+      headers,
+      credentials: "same-origin",
```

Comment:
1. I think `same-origin` is the default now
2. "Credentials" here refers to Cookies, which we aren't using
3. We won't be on the same origin anyways, since we are a web extension.

I think we should remove this.